extension/main.js

```diff
@@ -16,31 +16,51 @@
 	}
 
 	function update() {
-		gitHubNotifCount(function (count) {
+		gitHubNotifCount(function (count, interval) {
+			var period = 1;
+			if (interval !== parseInt(GitHubNotify.settings.get('interval'), 10)) {
+				GitHubNotify.settings.set('interval', interval);
+				period = Math.ceil(interval / 60);
+				if (period < 1) {
+					period = 1;
+				}
+                chrome.alarms.create({when: Date.now() + 2000 + (period * 60 * 1000)});
```

Comment:
OK I tried moving this down by one line, and that fixes the weird browser-cache issue, now each request hits the server.

However, I'm still getting strange results, whereby the counter is still stuck at 1, even though I have no unread notifications. I get 304s from the API (and they don't even affect the rate limit — hooray for that!), but the browser backfills the request with what seems to be bad content.